packages/webpack/mixins/esbuild/mixin.core.js

```diff
@@ -0,0 +1,85 @@
+const { Mixin } = require('hops-mixin');
+const importComponentTransform = require.resolve(
+  './importComponentTransform.js'
+);
+
+const isLoaderFor = (file) => (test) => {
```

Comment:
I'd use `fileType` here instead of `file` to make the purpose more obvious.

Reply:
I think it's fine to only handle `RegExp | Regexp[]` for now. But maybe it makes sense to warn, if it's a string or an object. So people with custom loader configurations know, that certain parts of their code (or the complete code) won't be bundled with esbuild.

Reply:
> I'd use fileType here instead of file to make the purpose more obvious.

The test runs against a file though (AFAICT)

>  So people with custom loader configurations know, that certain parts of their code (or the complete code) won't be bundled with esbuild.

I think it is obvious that esbuild will only transpile JavaScript and TypeScipt files. Getting warnings for e.g. `.css` because I am using the `--experimentalEsbuild` flag will create confusion IMO.

Reply:
> The test runs against a file though (AFAICT)

Yes, but the `file`-argument represents a pattern — a string ending with either `.js` or `.ts`. There won't ever be passed a file path into that function during the runtime of Webpack. That's why the name is not optimal.

Reply:
I've applied your second suggestion but I would like to leave the `file` thingy as is.